sympy/parsing/c/c_parser.py

```diff
@@ -261,8 +290,14 @@ def transform_var_decl(self, node):
                         else:
                             value = val
 
-                    elif (node.type.kind == cin.TypeKind.FLOAT):
-                        type = FloatBaseType(String('real'))
+                    elif node.type.kind in supported_float_types:
+                        if node.type.kind == cin.TypeKind.FLOAT:
+                            type = FloatBaseType(String('real'))
```

Comment:
you can import `real` from codegen instead of declaring it here again

Reply:
Yeah, right :sweat_smile:

Reply:
But, I am confused among `real` and `float32`, `intc` and `integer`. Which should I choose? According to the comments, for `real` and `integer`, precision will be chosen by the code printers!

Reply:
Once we decide which one to use in these cases(whether `real` or `float32` in case of `float` and `integer` or `intc` or `int32` in case of `int`), the other code(e.g.; `parm_decl` and `func_decl`), which has `IntBaseType` and `FloatBaseType`, also will have to be updated I guess

Reply:
We can use `intc` and `float32` for now then, and you can update the other visitors accordingly if it's required.

Reply:
Hi, I found a shortcoming of `parm_decl` while updating the data types there. It is not possible to get the default value of parameters from python bindings!